crates/bevy_ui/src/widget/text.rs

```diff
@@ -1,49 +1,204 @@
+///
+/*
+```
+# use bevy_asset::Handle;
+# use bevy_color::Color;
+# use bevy_color::palettes::basic::BLUE;
+# use bevy_ecs::World;
+# use bevy_text::{Font, JustifyText, TextBlock, TextStyle};
+# use bevy_ui::Text;
+#
+# let font_handle: Handle<Font> = Default::default();
+# let mut world = World::default();
+#
+// Basic usage.
+world.spawn(Text::new("hello world!"));
+
+// With non-default style.
+world.spawn((
+    Text::new("hello world!"),
+    TextStyle {
+        font: font_handle.clone().into(),
+        font_size: 60.0,
+        color: BLUE.into(),
+    }
+));
+
+// With text justification.
+world.spawn((
+    Text::new("hello world\nand bevy!"),
+    TextBlock::new_with_justify(JustifyText::Center)
+));
+```
+*/
+#[derive(Component, Debug, Default, Clone, Deref, DerefMut, Reflect)]
+#[reflect(Component, Default, Debug)]
+#[require(
+    TextBlock,
+    TextStyle,
+    TextNodeFlags,
+    Node,
+    Style, // TODO: Remove when Node uses required components.
+    ContentSize, // TODO: Remove when Node uses required components.
+    FocusPolicy, // TODO: Remove when Node uses required components.
+    ZIndex, // TODO: Remove when Node uses required components.
+    BackgroundColor, // TODO: Remove when Node uses required components.
+    Visibility, // TODO: Remove when Node uses required components.
+    Transform // TODO: Remove when Node uses required components.
+)]
+pub struct TextNEW(pub String);
+
+impl TextNEW {
+    /// Makes a new text component.
+    pub fn new(text: impl Into<String>) -> Self {
+        Self(text.into())
+    }
+}
+
+impl TextSpanAccess for TextNEW {
+    fn read_span(&self) -> &str {
+        self.as_str()
+    }
+    fn write_span(&mut self) -> &mut String {
+        &mut *self
+    }
+}
+
+impl From<&str> for TextNEW {
+    fn from(value: &str) -> Self {
+        Self(String::from(value))
+    }
+}
+
+impl From<String> for TextNEW {
+    fn from(value: String) -> Self {
+        Self(value)
+    }
+}
+
+/// A span of UI text in a tree of spans under an entity with [`Text`].
+///
+/// Spans are collected in hierarchy traversal order into a [`ComputedTextBlock`] for layout.
+///
+/*
+```
+# use bevy_asset::Handle;
+# use bevy_color::Color;
+# use bevy_color::palettes::basic::{RED, BLUE};
+# use bevy_ecs::World;
+# use bevy_text::{Font, TextStyle, TextSection};
+# use bevy_ui::{Text, TextSpan};
+#
+# let font_handle: Handle<Font> = Default::default();
+# let mut world = World::default();
+#
+world.spawn((
+    Text::new("Hello, "),
+    TextStyle {
+        font: font_handle.clone().into(),
+        font_size: 60.0,
+        color: BLUE.into(),
+    }
+))
+.with_child((
+    TextSpan::new("World!"),
+    TextStyle {
+        font: font_handle.into(),
+        font_size: 60.0,
+        color: RED.into(),
+    }
+));
+```
+*/
+#[derive(Component, Debug, Default, Clone, Deref, DerefMut, Reflect)]
+#[reflect(Component, Default, Debug)]
+#[require(TextStyle, GhostNode, Visibility(hidden_visibility))]
```

Comment:
Like with `TextSpan2d` why do we require `Visibility` here I don't think it's very useful to set Visibility for individual text spans and it doesn't seem like that's supported anyway.

Reply:
Oh right, `GhostNode` already requires `Visibility` and `Transform`.